### ISSUE TITLE
Has this been tested for Python 3.11?

### ISSUE THREAD

**Issue by richardstardust** [janelia-pypi/mightex_device_python#1]
I was happy to find this library since it would same me some work. Unfortunately it does not work with Python 3.11.
Some testing showed that there a lot of small changes that probably have to made to make it work.
If you would have time to update it I would be very happy and it would be a nice contribution to the Mightex community.

Best regards,
/Richard

**Comment by peterpolidoro** [comment]
Hi,

I have not used this library in years and I have not yet had a chance to test out Python 3.11. Do you happen to have some examples of changes that I would need to make in order for it to work for you?

**Comment by richardstardust** [comment]
I am not good at Python but with some trial and error I change these (most things have to do with strings and bytes I guess):
I commented out "from exceptions import Exception" since you can use Exception without import.
response = response.replace('#','')      ->      response = response.replace(b'#',b'')
response_list = response.split(' ')      ->      response_list = response.split(b' ')
if '#!' in response:      ->      if b'#!' in response:
elif '#?' in response:      ->      elif b'#?' in response:
found_list = p.findall(device_info)      ->      found_list = p.findall(device_info.decode())
if response == b'0':      ->     if response == b'0':
...
raise MightexError('Unknown response: {0}'.format(response))      ->    raise MightexError('Unknown response: {0}'.format(response.decode()))
response = response.replace(channel_str,'')      ->      response = response.replace(channel_str.encode(),b'')

Then there were some "channel = int(channel) + 1" missing

**Comment by richardstardust** [comment]
Or maybe the strings and bytes problem should be fixed in "serial_device_python" instead.

**Comment by peterpolidoro** [comment]
Yes I have fixed this in some of my other Python libraries. I will modify this one too when I have a chance, but it might take me a week or two. Unfortunately I no longer have any mightex devices to test code changes on though.

**Comment by richardstardust** [comment]
I can do some of the testing with my Mightex device, like running your example code or if you have some other test code.

**Comment by richardstardust** [comment]
It seems I change something in the wrong place so I reinstalled everything and did a new try to.
This time I changed the following which probably is a little better:

Line 8 in mightex_device.py
#from exceptions import Exception

changed _args_to_request
def _args_to_request(self,*args):
    request = ' '.join(map(str,args))
    request = request + '\n\r'
    return request.encode()

changed _send_request_get_response since it seemed unnecessary to me that _args_to_request and _debug_print is called twice for every command and then I added the decode() for the response
    def _send_request_get_response(self,request):
        '''
        Sends request to device over serial port and
        returns response.
        '''
        response = ''
        lock_acquired = self._lock.acquire()
        try:
            # request = self._args_to_request(*args)
            # self._debug_print('request', request)
            response = self._serial_device.write_read(request,use_readline=True,check_write_freq=False,delay_write=True)
            response = response.decode()


changed readline() to readline().decode() on the two places where it is used

Other changes:
It seems to me that the folowing functions were missing the +1 to the channel number:
def get_strobe_profile(self,channel):
def set_strobe_profile(self,channel,profile):
def get_trigger_profile(self,channel):

channel = int(channel) + 1

added delay to the reset function since my Jupyter notebook hanged without it:
    def reset(self):
        '''
        Soft reset device.
        '''
        request = self._args_to_request('Reset')
        self._debug_print('request', request)
        self._send_request_get_response(request)
        time.sleep(0.1)
